Source/RapyutaSimulationPlugins/Private/Tools/SimulationState.cpp

```diff
@@ -240,32 +242,70 @@ void ASimulationState::AttachSrv(UROS2GenericSrv* Service)
+    spawnableComponent->RegisterComponent();
+    spawnableComponent->InitializeParameters(InROSSpawnRequest);
+    newEntity->AddInstanceComponent(spawnableComponent);
+    newEntity->Rename(*InROSSpawnRequest.StateName);
+#if WITH_EDITOR
+    newEntity->SetActorLabel(*InROSSpawnRequest.StateName);
+#endif
+
+    // Add tags
+    for (auto& tag : InROSSpawnRequest.Tags)
+    {
+        newEntity->Tags.Emplace(tag);
+    }
+
+    // Finish spawning Entity
+    UGameplayStatics::FinishSpawningActor(newEntity, InEntityTransform);
+
+    // Add to [Entities]
+    AddEntity(newEntity);
+    return newEntity;
+}
+
 void ASimulationState::SpawnEntitySrv(UROS2GenericSrv* Service)
 {
     UROS2SpawnEntitySrv* SpawnEntityService = Cast<UROS2SpawnEntitySrv>(Service);
 
-    FROSSpawnEntityRequest Request;
-    SpawnEntityService->GetRequest(Request);
+    FROSSpawnEntityRequest spawnEntityRequest;
+    SpawnEntityService->GetRequest(spawnEntityRequest);
 
-    FROSSpawnEntityResponse Response;
-    Response.bSuccess = CheckSpawnableEntity(Request.Xml, false) && CheckEntity(Request.StateReferenceFrame, true);
+    FROSSpawnEntityResponse spawnEntityResponse;
+    spawnEntityResponse.bSuccess =
+        CheckSpawnableEntity(spawnEntityRequest.Xml, false) && CheckEntity(spawnEntityRequest.StateReferenceFrame, true);
 
-    if (Response.bSuccess)
+    if (spawnEntityResponse.bSuccess)
     {
-        const FString& entityModelName = Request.Xml;
-        const FString& entityName = Request.StateName;
+        const FString& entityModelName = spawnEntityRequest.Xml;
+        const FString& entityName = spawnEntityRequest.StateName;
         verify(false == entityName.IsEmpty());
         if (nullptr == URRUObjectUtils::FindActorByName<AActor>(GetWorld(), entityName))
         {
             // Calculate to-be-spawned entity's [world transf]
-            FVector relLocation(Request.StatePosePositionX, Request.StatePosePositionY, Request.StatePosePositionZ);
-            FTransform relativeTransf = URRConversionUtils::TransformROSToUE(FTransform(Request.StatePoseOrientation, relLocation));
+            FVector relLocation(spawnEntityRequest.StatePosePositionX,
+                                spawnEntityRequest.StatePosePositionY,
+                                spawnEntityRequest.StatePosePositionZ);
+            FTransform relativeTransf =
+                URRConversionUtils::TransformROSToUE(FTransform(spawnEntityRequest.StatePoseOrientation, relLocation));
             FTransform worldTransf;
-            URRGeneralUtils::GetWorldTransform(
-                Request.StateReferenceFrame,
-                Entities.Contains(Request.StateReferenceFrame) ? Entities[Request.StateReferenceFrame] : nullptr,
-                relativeTransf,
-                worldTransf);
+            URRGeneralUtils::GetWorldTransform(spawnEntityRequest.StateReferenceFrame,
+                                               Entities.FindRef(spawnEntityRequest.StateReferenceFrame),

```

Comment:
Does FindRef return null if entities do not have item?

Reply:
The name could be abit mis leading, since it actually returns a value, not a reference.
As defined [here](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Containers/Map.h#L673), it returns default value of `AActor*`, which should be [zero](https://godbolt.org/z/sn8TqGeah).
A quick check could help verify it:
```
    TMap<const TCHAR*, AActor*> testMap = {{TEXT("a"), this}};
    verify(nullptr == testMap.FindRef(TEXT("b")));
```